appender_test.go

```diff
@@ -698,98 +700,94 @@ func TestAppenderFlushBytes(t *testing.T) {
 }
 
 func TestAppenderFlushRequestError(t *testing.T) {
+	originalError := []byte(`{"error": {"root_cause": [{"type": "x_content_parse_exception","reason": "reason"}],"type": "x_content_parse_exception","reason": "reason","caused_by": {"type": "json_parse_exception","reason": "reason"}},"status": 400}`)
+	reducedError := []byte(`{"error":{"type":"x_content_parse_exception","caused_by":{"type":"json_parse_exception"}}}`)
+
 	// This test ensures that the appender correctly categorizes and quantifies
 	// failed requests with different failure scenarios. Since a bulk request
 	// contains N documents, the appender should increment the categorized
 	// failure by the same number of documents in the request.
-	test := func(t *testing.T, sc int, errMsg string) {
-		var bytesTotal int64
-		var bytesUncompressedTotal int64
-		client := docappendertest.NewMockElasticsearchClient(t, func(w http.ResponseWriter, r *http.Request) {
-			bytesTotal += r.ContentLength
-			_, _, stat := docappendertest.DecodeBulkRequestWithStats(r)
-			bytesUncompressedTotal += stat.UncompressedBytes
-			w.WriteHeader(sc)
-			w.Write([]byte(`{"error": {"root_cause": [{"type": "x_content_parse_exception","reason": "reason"}],"type": "x_content_parse_exception","reason": "reason","caused_by": {"type": "json_parse_exception","reason": "reason"}},"status": 400}`))
-		})
+	for _, includeSource := range []docappender.Value{docappender.Unset, docappender.True} {
```

Comment:
non-blocker: I don't think we need to intentionally leave out docappender.False here. docappender.False will be the most used config and if we include it here, at least we know we're returning errors as-is for False and True.